tests/gha/test_mr_generate_summary.py

```diff
@@ -0,0 +1,94 @@
+from pathlib import Path
```

Comment:
Awesome that we start testing this code as well! 🚀 

Wondering if it's worth separating these "infrastructure" tests from the production code tests. In my opinion it should be a separate job / workflow in the GitHub actions to clearly separate things 🤔 Treating this as the regular production tests also makes these infrastructure tests a dependency for CI steps which target the production code which seems off to me 🤔

Reply:
We could move the test files from `tests/gha` to `.github/tests`. Would this be what you have in mind?

I fear that when splitting `.github/workflows/continous-integration.yml` s.t. `test-gh-actions` ran in a new separate workflow, we would duplicate a lot of CI code. But maybe this is worth to break the dependency?

Reply:
> We could move the test files from tests/gha to .github/tests. Would this be what you have in mind?

That sounds good 👍🏻 

> we would duplicate a lot of CI code.

Is it actually? These tests are pretty lightweight so all we'd need is 
1. Check out source code
2. Install dependencies
3. Test with a single Python version

Isn't it? 
Curious to hear @joejuzl 's opinion on it

Reply:
Agree it makes sense to separate from production code tests 👍

Reply:
Tests are now separated